src/api/common/MessageDispatcher.ts

```diff
@@ -123,19 +123,26 @@ export class MessageDispatcher<OutgoingRequestType extends string, IncomingReque
 
 	handleMessage(message: Message<IncomingRequestType>) {
 		if (message.type === "response") {
-			this._messages[message.id].resolve(message.value)
-
-			delete this._messages[message.id]
+			const pendingRequest = this._messages[message.id]
+			if (pendingRequest != null) {
+				pendingRequest.resolve(message.value)
+				delete this._messages[message.id]
+			} else {
+				console.warn(`Unexpected response: ${message.id} (was the page reloaded?)`)
+			}
 		} else if (message.type === "requestError") {
-			this._messages[message.id].reject(objToError(downcast(message).error))
-
-			delete this._messages[message.id]
-		} else {
+			const pendingRequest = this._messages[message.id]
+			if (pendingRequest != null) {
+				pendingRequest.reject(objToError(message.error))
+				delete this._messages[message.id]
+			} else {
+				console.warn(`Unexpected error response: ${message.id} (was the page reloaded?)`)
+			}
```

Comment:
This `retrieve pending, then (do an action and delete) or (warn)` pattern seems like it could be put into a method (handlePendingRequestIfPresent with a thunk parameter maybe?)

Reply:
I considered that but decided that it's not worth it for just these two cases with different error message

Reply:
Oh, the error message is different. I we could probably have used the same one for a warning - but I guess the current code is fine if it only repeats twice anyway